test/e2e/toolbox_test.go

```diff
@@ -0,0 +1,346 @@
+
+		session = podmanTest.Podman([]string{"start", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"exec", "test",
+			"df", "-h", "/dev/shm"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		lines = session.OutputToStringArray()
+		fields = strings.Fields(lines[len(lines)-1])
+		containerShmSize, err = strconv.Atoi(re.FindString(fields[1]))
+		Expect(err).To(BeNil())
+
+		Expect(hostShmSize).To(BeNumerically("~", containerShmSize, 1))
+	})
+
+	It("podman create --userns=keep-id --user root:root - entrypoint - entrypoint is executed as root", func() {
+		var session *PodmanSessionIntegration
+
+		session = podmanTest.Podman([]string{"run", "--userns=keep-id", "--user", "root:root", ALPINE,
+			"id"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		ok, _ := session.GrepString("uid=0(root) gid=0(root)")
+		Expect(ok).To(BeTrue())
+	})
+
+	It("podman create --userns=keep-id + podman exec - correct names of user and group", func() {
+		var session *PodmanSessionIntegration
+		var err error
+
+		currentUser, err := user.Current()
+		Expect(err).To(BeNil())
+
+		currentGroup, err := user.LookupGroupId(currentUser.Gid)
+		Expect(err).To(BeNil())
+
+		expectedOutput := fmt.Sprintf("uid=%s(%s) gid=%s(%s)",
+			currentUser.Uid, currentUser.Username,
+			currentGroup.Gid, currentGroup.Name)
+
+		fmt.Printf("Expected output: %s\n", expectedOutput)
+
+		session = podmanTest.Podman([]string{"create", "--name", "test", "--userns=keep-id", ALPINE,
+			"sleep", "1000"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		Expect(err).To(BeNil())
+
+		session = podmanTest.Podman([]string{"start", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"exec", "test",
+			"id"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		ok, _ := session.GrepString(expectedOutput)
+		Expect(ok).To(BeTrue())
+	})
+
+	It("podman create --userns=keep-id - entrypoint - adding user with useradd and then removing his password", func() {
+		var session *PodmanSessionIntegration
+
+		var username string = "testuser"
+		var homeDir string = "/home/testuser"
+		var shell string = "/bin/sh"
+		var uid string = "1001"
+		var gid string = "1001"
+
+		session = podmanTest.Podman([]string{"create", "--name", "test", "--userns=keep-id", "--user", "root:root", fedoraToolbox, "sh", "-c",
+			fmt.Sprintf("useradd --home-dir %s --shell %s --uid %s %s; passwd --delete %s; sleep 1000",
+				homeDir, shell, uid, username, username)})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"start", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		time.Sleep(3 * time.Second)
```

Comment:
I see a lot of these sleeps. Tests are already close to the CI timeout limit, and every second counts. Are these necessary? If so, would 1 second be OK? (And can you document in a comment why they're necessary?)

Reply:
I agree that the timeout is too long. My motivation was to give the entry-point enough time to be executed. It definitely can be just a second or even less. But I've got a different idea of how to go about this. Instead of explicitly waiting I switched the invocation of `podman logs` with `podman exec` that happens quite often. Actually... now I'm wondering if that will fix it :sweat_smile: .

Anyway, the timeout can be definitely lower but some is definitely necessary.